feat: introduce `twoslash-svelte`

Hugos68 opened the pull request:
This PR attempts to create `twoslash-svelte` which is an extension of `twoslash`that supports [Svelte](https://github.com/sveltejs/svelte).

Hugos68 committed:
Initial efforts

Hugos68 committed:
Fix readme

Hugos68 committed:
fix lockfile

Hugos68 committed:
remove comment

Hugos68 committed:
It works!

Hugos68 committed:
Fixed generateSourceMap

Hugos68 committed:
Added doc page

Hugos68 commented:
@antfu I'm unable to configure the docsite to use `twoslash-svelte`. I don't have much experience with shiki and/or vitepress. A helping hand would he appreciated.

Hugos68 committed:
stuff works

Hugos68 committed:
Yay it works

Hugos68 committed:
Fix stuff

Hugos68 committed:
doc issues

Hugos68 committed:
Added TODO

Hugos68 committed:
docs now run

Hugos68 committed:
Moved package version to pnpm catalog

Hugos68 committed:
fix lockfile

Hugos68 committed:
Hopefully fix lockfile stuff

Hugos68 committed:
test

Hugos68 committed:
hopefully finally fix shit

Hugos68 committed:
Added note

Hugos68 committed:
Better example

Hugos68 committed:
Fix type issue

Hugos68 committed:
Added query marker to example

Hugos68 commented:
@antfu @orta
A review would be appreiciated, I'm now at a point where I feel confident I have the code working. I was able to copy a big portion from `twoslash-vue` as it works very similarly (convert source to `tsx` then typecheck and use sourcemaps to put markers).

All that's left is tests.

orta commented:
From a high level, this looks very cool to me!

Hugos68 committed:
added back conflict removal

Hugos68 committed:
Added svelte2tsx as peer dep

Hugos68 commented:
There seems to be a bug related to the `$` inside `twoslash`. Or `@shikijs/twoslash-vitepress`. Everytime a variable has the `$` inside of it there are 2 popups:

1: ![image](https://github.com/user-attachments/assets/76daebfa-af95-4448-a0d4-538fead8c8da)
2: ![image](https://github.com/user-attachments/assets/d351fcf5-1c19-4b32-bb6f-eb6a7df90ecb)

As you can see both popups show information about `$state` but one is at the first index of the word and the other is at the middle index (which is the correct one). Why is there an extra popup for the `$`?

In the HTML output you can see that instead of a single `span` the `$` and `state` are seperated:
![image](https://github.com/user-attachments/assets/ede97c92-00f7-4318-9080-76b692a44572)

Hugos68 committed:
cleanup

Hugos68 committed:
Cleanup

Hugos68 commented:
Also, for some reason `twoslash-vue` test aren't passing anymore, I have no clue why as I didn't touch the tests from `twoslash-vue`.

Hugos68 committed:
Added fixture tests for Svelte

Hugos68 committed:
perhaps fixed lockfile

Hugos68 committed:
idk

Hugos68 committed:
fix lockfile

Hugos68 committed:
Added cut block tests

Hugos68 committed:
cleanup

Hugos68 committed:
Removed windows line endings in fixtures

Hugos68 committed:
Remove unnecessary dev dependency

Hugos68 committed:
Remove redundant dep (use positionConverter instead)

Hugos68 committed:
remove space

Hugos68 committed:
Fixed Vue test

Hugos68 committed:
Revert "Fixed Vue test"

This reverts commit 31efdc96712810534ed01b264e5bd42cba6d3b9b.

Hugos68 committed:
Simpler tests

Hugos68 committed:
Embed types during build

Hugos68 committed:
Fixed it!

Hugos68 committed:
Fixed it maybe again?

Hugos68 committed:
Fixed it perhaps again?

Hugos68 commented:
@dominikg, I might need some help as you're one of the maintainers of `svelte2tsx` correct?
When compiling the `tsx` coming from `svelte2tsx` how can I include the `.d.ts` files? I've tried many methods but can't find a consistent way where the user can install `svelte2tsx` and the `twoslash-svelte` package can use those files.

dominikg commented:
> @dominikg, I might need some help as you're one of the maintainers of `svelte2tsx` correct? 
sorry for not responding sooner. I do not maintain svelte2tsx, @dummdidumm would be the best person to reach out to, you can try `#editor-tools-questions` on https://svelte.dev/chat

Hugos68 commented:
No problem, and thanks, will do!

Hugos68 committed:
Merge branch 'main' into feat/svelte

Hugos68 committed:
fix lint issue

Hugos68 committed:
fix docs not deploying

Hugos68 committed:
tweak

Hugos68 commented:
@antfu This is ready for review

PuruVJ commented:
Side comment, do remember to add svelte to dependencies or peers dependencies or devDependency :)

Hugos68 commented:
@PuruVJ Good call! Thanks, will do.

antfu committed:
Merge main into feat/svelte

antfu committed:
chore: improvements and fixes for twoslash-svelte

- bump version to match monorepo
- fix variable naming for source vs generated converters
- await vitest snapshots
- export alias `createTwoslasherSvelte`
- update snapshots

antfu committed:
Merge origin/main into feat/svelte and resolve pnpm lockfile conflict

antfu committed:
chore: update failing snapshots after pnpm 11 merge

antfu committed:
fix: ci failures and improve test coverage
